decode_test.go

```diff
@@ -977,14 +978,21 @@ func (s *S) TestUnmarshalSliceOnPreset(c *C) {
 }
 
 func (s *S) TestUnmarshalStrict(c *C) {
-	v := struct{ A, B int }{}
+	v := struct {
+		A, B int
+		Z    map[string]string
+	}{}
 
 	err := yaml.UnmarshalStrict([]byte("a: 1\nb: 2"), &v)
 	c.Check(err, IsNil)
 	err = yaml.Unmarshal([]byte("a: 1\nb: 2\nc: 3"), &v)
 	c.Check(err, IsNil)
 	err = yaml.UnmarshalStrict([]byte("a: 1\nb: 2\nc: 3"), &v)
-	c.Check(err, ErrorMatches, "yaml: unmarshal errors:\n  line 1: field c not found in struct struct { A int; B int }")
+	c.Check(err, ErrorMatches, "yaml: unmarshal errors:\n  line 1: field c not found in struct struct { A int; B int; Z map\\[string\\]string }")
+	err = yaml.UnmarshalStrict([]byte("a: 1\na: 2"), &v)
+	c.Check(err, ErrorMatches, "yaml: unmarshal errors:\n  line 1: field a already set in struct struct { A int; B int; Z map\\[string\\]string }")
+	err = yaml.UnmarshalStrict([]byte("z:\n  a: 1\n  a: 1"), &v)
+	c.Check(err, ErrorMatches, "yaml: unmarshal errors:\n  line 2: field a already set in map")
```

Comment:
Let's have a test for unmarshaling duplicate values into a struct with inline members too, please (including an inline struct and a map)

Reply:
Added, I think - I hadn't seen this form before so not 100% confident.